packages/website/docs/Internal/Grammar.md

```diff
@@ -0,0 +1,53 @@
+---
+title: Grammar
+urlcolor: blue
+author:
+  - Quinn Dougherty
+---
+
+Formal grammar specification, reference material for parser implementation.
+
+_In all likelihood the reference will have to be debugged as we see what tests pass and don't pass during implementation_.
+
+## Lexical descriptions of constants and identifiers
+
+- `<number>` ::= `[-]? [0-9]+ (.[0-9]+)? | [-]? [0-9]+ (.[0-9]+)? [e] [-]? [0-9]+`
+- `<symbol>` ::= `[a-zA-Z]+ [a-zA-Z0-9]?`
+- `<baseDistribution>` ::= `(normal(<number>, <number>) | beta(<number>, <number>) | triangular(<number>, <number>, <number>) | cauchy(<number>, <number>) | lognormal(<number>, <number>)) | uniform(<number>, <number>) | <number> to <number> | exponential(<number>)`
```

Comment:
Most of these just seem like functions to me, I don't see why they need to be hardcoded here.

Reply:
It's because the parser implementation requires that you specify them as dedicated symbols.

Reply:
This brings me back to the question about the functions. https://github.com/quantified-uncertainty/squiggle/pull/338/files#r856284344

I still don't see why the parser needs to know our specific available functions. (Especially because users can define new functions, and I don't think the grammar can tell the difference)